web3.js/test/transaction.test.ts

```diff
@@ -962,4 +963,48 @@ describe('Transaction', () => {
     t1.partialSign(signer);
     t1.serialize();
   });
+
+  it('externally signed versioned transaction', () => {
+    const signer = Keypair.generate();
+    const invalidSigner = Keypair.generate();
+
+    const recentBlockhash = new PublicKey(3).toBuffer();
+
+    const message = new TransactionMessage({
+      payerKey: signer.publicKey,
+      instructions: [
+        new TransactionInstruction({
+          data: Buffer.from('Hello!'),
+          keys: [
+            {
+              pubkey: signer.publicKey,
+              isSigner: true,
+              isWritable: true,
+            },
+          ],
+          programId: new PublicKey(
+            'MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr',
+          ),
+        }),
+      ],
+      recentBlockhash: bs58.encode(recentBlockhash),
+    });
+
+    const transaction = new VersionedTransaction(message.compileToV0Message());
+
+    const signature = sign(transaction.message.serialize(), signer.secretKey);
+
+    transaction.addSignature(signer.publicKey, signature);
+
+    expect(transaction.signatures).to.have.length(1);
+    expect(transaction.signatures[0]).to.eq(signature);
+    expect(() => {
+      transaction.addSignature(signer.publicKey, new Uint8Array(32));
+    }).to.throw('Signature must be 64 bytes long');
+    expect(() => {
+      transaction.addSignature(invalidSigner.publicKey, new Uint8Array(64));
+    }).to.throw(
+      `Can not add signature; \`${invalidSigner.publicKey.toBase58()}\` is not required to sign this transaction`,
+    );
```

Comment:
For sure want these to be separate tests, so that when they fail, the reader doesn't have to dig to find out what functionality was broken.

```typescript
describe('addSignature', () => {
  it('appends an externally generated signature at the correct index', () => { /* ... */ })
  it('fatals when the signature is the wrong length', () => { /* ... */ })
  it('fatals when adding a signature for a public key that has not been marked as a signer', () => { /* ... */ })
});